packages/api-key-service/src/functions/paginate.ts

```diff
@@ -0,0 +1,49 @@
+import { ApiKeyModel } from '@/models'
+import { httpHandler, UnauthorizedException } from '@homeservenow/serverless-aws-handler'
+import { batchGetApiKeys } from './../services'
+import { connectSessionVerifyDecodeIdTokenWithPublicKeys, LoginIdentity } from '@reapit/connect-session'
+import publicKeys from './../../publicKeys.json'
+
+type Pagintation<T> = {
+  items: T[]
+  meta: {
+    // count: number
+    nextCursor: string
+  }
+}
+
+export const paginateApiKeys = httpHandler<void, Pagintation<ApiKeyModel>>({
+  handler: async ({ event }): Promise<Pagintation<ApiKeyModel>> => {
+    let customer: LoginIdentity | undefined
+
+    try {
+      customer = await connectSessionVerifyDecodeIdTokenWithPublicKeys(
+        event.headers?.Authorization as string,
+        process.env.CONNECT_USER_POOL as string,
+        publicKeys,
+      )
+
+      if (typeof customer === 'undefined' || !customer.developerId) {
+        throw new Error('Unauthorised')
+      }
+    } catch (e) {
+      throw new UnauthorizedException(e.message)
+    }
+
+    const response = await batchGetApiKeys(
+      customer as LoginIdentity & { developerId: string },
+      event?.queryStringParameters?.nextCursor ? { id: event?.queryStringParameters?.nextCursor } : undefined,
+    )
+
+    const pagination: Pagintation<ApiKeyModel> = {
+      items: [],
+      meta: response[1],
```

Comment:
Will response always be defined / have length? Looks like could throw an exception if not... do correct me if I'm wrong!

Reply:
I'm not sure? It's suppose to be an async iterator which is weird to me so will research 👍

Reply:
response[0] and response[1] will also exist though! Only response[0] may have a different type depending on response

Reply:
Cool, just checking